dyna_precice.c

```diff
@@ -0,0 +1,2177 @@
+/*     published by the Free Software Foundation(version 2);    */
+/*                    */
+
+/*     This program is distributed in the hope that it will be useful,   */
+/*     but WITHOUT ANY WARRANTY; without even the implied warranty of    */ 
+/*     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the      */
+/*     GNU General Public License for more details.                      */
+
+/*     You should have received a copy of the GNU General Public License */
+/*     along with this program; if not, write to the Free Software       */
+/*     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.         */
+
+#include <stdio.h>
+#include <math.h>
+#include <stdlib.h>
+#include <string.h>
+#include "CalculiX.h"
+
+#ifdef SPOOLES
+#include "spooles.h"
+#endif
+#ifdef SGI
+#include "sgi.h"
+#endif
+#ifdef TAUCS
+#include "tau.h"
+#endif
+#ifdef PARDISO
+#include "pardiso.h"
+#endif
+#ifdef PASTIX
+#include "pastix.h"
+#endif
+
+/* Adapter: Add header */
+#include "adapter/PreciceInterface.h"
+
+void dyna_precice(double **cop,ITG *nk,ITG **konp,ITG **ipkonp,char **lakonp,ITG *ne,
+	  ITG **nodebounp,ITG **ndirbounp,double **xbounp,ITG *nboun,
+	  ITG **ipompcp,ITG **nodempcp,double **coefmpcp,char **labmpcp,
+	  ITG *nmpc,ITG *nodeforc,ITG *ndirforc,double *xforc,
+	  ITG *nforc,ITG *nelemload,char *sideload,double *xload,
+	  ITG *nload,
+	  ITG **nactdofp,ITG *neq,ITG *nzl,ITG *icol,ITG *irow,
+	  ITG *nmethod,ITG **ikmpcp,ITG **ilmpcp,ITG **ikbounp,
+	  ITG **ilbounp,double *elcon,ITG *nelcon,double *rhcon,
+	  ITG *nrhcon,double *cocon,ITG *ncocon,
+	  double *alcon,ITG *nalcon,double *alzero,
+	  ITG **ielmatp,ITG **ielorienp,ITG *norien,double *orab,
+	  ITG *ntmat_,double **t0p,
+	  double **t1p,ITG *ithermal,double *prestr,ITG *iprestr,
+	  double **voldp,ITG *iperturb,double **stip,ITG *nzs,
+	  double *timepar,double *xmodal,
+	  double **veoldp,char *amname,double *amta,
+	  ITG *namta,ITG *nam,ITG *iamforc,ITG *iamload,
+	  ITG **iamt1p,ITG *jout,
+	  ITG *kode,char *filab,double **emep,double *xforcold,
+	  double *xloadold,
+	  double **t1oldp,ITG **iambounp,double **xbounoldp,ITG *iexpl,
+	  double *plicon,ITG *nplicon,double *plkcon,ITG *nplkcon,
+	  double **xstatep,ITG *npmat_,char *matname,ITG *mi,
+	  ITG *ncmat_,ITG *nstate_,double **enerp,char *jobnamec,
+	  double *ttime,char *set,ITG *nset,ITG *istartset,
+	  ITG *iendset,ITG **ialsetp,ITG *nprint,char *prlab,
+	  char *prset,ITG *nener,double *trab,
+	  ITG **inotrp,ITG *ntrans,double **fmpcp,ITG *ipobody,ITG *ibody,
+	  double *xbody,ITG *nbody,double *xbodyold,ITG *istep,
+	  ITG *isolver,ITG *jq,char *output,ITG *mcs,ITG *nkon,
+	  ITG *mpcend,ITG *ics,double *cs,ITG *ntie,char *tieset,
+	  ITG *idrct,ITG *jmax,
+	  double *ctrl,ITG *itpamp,double *tietol,ITG *nalset,
+	  ITG *ikforc,ITG *ilforc,double *thicke,ITG *nslavs,ITG *nmat,
+	  char *typeboun,ITG *ielprop,double *prop,char *orname,
+	  double *t0g,double *t1g
```

Comment:
```suggestion
	  double *t0g,double *t1g,
```

There's a missing comma that prevents the build.